### PR TITLE
Mac wakeup workaround

### PR DESCRIPTION
**Pull request by fazakis**
Introduced a call to "caffeinate" as suggested by @bellwether-softworks

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -93,6 +93,14 @@ pub fn new() -> ServerPtr {
     Arc::new(RwLock::new(server))
 }
 
+fn mac_wakeup(){
+    #[cfg(target_os = "macos")]{
+        use std::process::Command;
+        Command::new("/usr/bin/caffeinate").arg("-u").arg("-t 5").spawn().expect("failed to execute caffeinate");
```
Comment (rustdesk) [inline src/server.rs]: `expect("")` may panic. please remove it. just use `ok()`;
https://doc.rust-lang.org/std/result/enum.Result.html#method.expect

File: src/server.rs [hunk 2]
```diff
@@ -93,6 +93,14 @@ pub fn new() -> ServerPtr {
...
+        println!("wake up macos");
```
Comment (rustdesk) [inline src/server.rs]: use `log::info!` instead of println! please.

File: src/server.rs (hunk 1 above)
Comment (fazakis) [reply src/server.rs]: Thanks

File: src/server.rs (hunk 2 above)
Comment (fazakis) [reply src/server.rs]: Thanks

File: src/server.rs [hunk 3]
```diff
@@ -252,6 +261,7 @@ async fn create_relay_connection_(
         ..Default::default()
     });
     stream.send(&msg_out).await?;
+    mac_wakeup();
```
Comment (rustdesk) [inline src/server.rs]: Why don't you put `mac_wakeup` in `create_tcp_connection`?

Comment (fazakis) [reply src/server.rs]: sorry almost blind

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/981

**Pull request merged**